ext/phar/phar_object.c

```diff
@@ -5370,8 +5405,8 @@ void phar_object_init(TSRMLS_D) /* {{{ */
 	phar_ce_archive = zend_register_internal_class(&ce TSRMLS_CC);
 	phar_ce_archive->ce_flags |= ZEND_ACC_FINAL_CLASS;
 
-	INIT_CLASS_ENTRY(ce, "PharData", php_archive_methods);
-	phar_ce_data = zend_register_internal_class(&ce TSRMLS_CC);
+	INIT_CLASS_ENTRY(ce, "PharData", php_archive_data_methods);
+	phar_ce_data = zend_register_internal_class_ex(&ce, phar_ce_archive, NULL  TSRMLS_CC);
 	phar_ce_data->ce_flags |= ZEND_ACC_FINAL_CLASS;
```

Comment:
When non-spl support Phar are not allow to create an instance object, this only allow static methods, they seems to only allow read only archive, Let me ask helly whether he knows the reason behind it.